docker_test.go

```diff
@@ -795,4 +1088,1001 @@ func consume(topics []string, master sarama.Consumer) (chan *sarama.ConsumerMess
+			  "review_body" : "Average product, expected much more."
+			},
+			"context": {
+				"ip": "14.5.67.21"
+			}
+		  }`, wdt.userId, uuid.Must(uuid.NewV4()).String()))
+			SendEvent(payloadTrack, "track", wdt.writeKey)
+		}
+	}
+
+	// Sending page event
+	if page, exists := wdt.whEventsCountMap["pages"]; exists {
+		for i := 0; i < page; i++ {
+			payloadPage := strings.NewReader(fmt.Sprintf(`{
+			"userId": "%s",
+			"messageId":"%s",
+			"type": "page",
+			"name": "Home",
+			"properties": {
+			  "title": "Home | RudderStack",
+			  "url": "http://www.rudderstack.com"
+			},
+			"context": {
+				"ip": "14.5.67.21",
+				"library": {
+					"name": "http"
+				}
+			  }
+		  }`, wdt.userId, uuid.Must(uuid.NewV4()).String()))
+			SendEvent(payloadPage, "page", wdt.writeKey)
+		}
+	}
+
+	// Sending screen event
+	if screen, exists := wdt.whEventsCountMap["screens"]; exists {
+		for i := 0; i < screen; i++ {
+			payloadScreen := strings.NewReader(fmt.Sprintf(`{
+			"userId": "%s",
+			"messageId":"%s",
+			"type": "screen",
+			"name": "Main",
+			"properties": {
+			  "prop_key": "prop_value"
+			},
+			"context": {
+				"ip": "14.5.67.21",
+				"library": {
+					"name": "http"
+				}
+			  }
+		  }`, wdt.userId, uuid.Must(uuid.NewV4()).String()))
+			SendEvent(payloadScreen, "screen", wdt.writeKey)
+		}
+	}
+
+	// Sending alias event
+	if alias, exists := wdt.whEventsCountMap["aliases"]; exists {
+		for i := 0; i < alias; i++ {
+			payloadAlias := strings.NewReader(fmt.Sprintf(`{
+			"userId": "%s",
+			"messageId":"%s",
+			"type": "alias",
+			"previousId": "name@surname.com",
+			"context": {
+				"traits": {
+				   "trait1": "new-val"
+				},
+				"ip": "14.5.67.21",
+				"library": {
+					"name": "http"
+				}
+            }
+		  }`, wdt.userId, uuid.Must(uuid.NewV4()).String()))
+			SendEvent(payloadAlias, "alias", wdt.writeKey)
+		}
+	}
+
+	// Sending group event
+	if group, exists := wdt.whEventsCountMap["groups"]; exists {
+		for i := 0; i < group; i++ {
+			payloadGroup := strings.NewReader(fmt.Sprintf(`{
+			"userId": "%s",
+			"messageId":"%s",
+			"type": "group",
+			"groupId": "groupId",
+			"traits": {
+			  "name": "MyGroup",
+			  "industry": "IT",
+			  "employees": 450,
+			  "plan": "basic"
+			},
+			"context": {
+				"traits": {
+					"trait1": "new-val"
+				},
+				"ip": "14.5.67.21",
+				"library": {
+					"name": "http"
+				}
+			}
+		  }`, wdt.userId, uuid.Must(uuid.NewV4()).String()))
+			SendEvent(payloadGroup, "group", wdt.writeKey)
+		}
+	}
+}
+
+// whDestinationTest Checking warehouse destination
+func whDestinationTest(t *testing.T, wdt *WareHouseDestinationTest) {
+	whGatewayTest(t, wdt)
+	whBatchRouterTest(t, wdt)
+	whTablesTest(t, wdt)
+}
+
+// whGatewayTest Checking for gateway jobs
+func whGatewayTest(t *testing.T, wdt *WareHouseDestinationTest) {
+	require.Contains(t, wdt.whEventsCountMap, "gateway")
+	gwEvents := wdt.whEventsCountMap["gateway"]
+
+	// Checking for the gateway jobs
```

Comment:
Instead of comments in the code you can do this, in case this test fails the printed logs could help with debugging.
```suggestion
	t.Log("Checking for the gateway jobs")
```